### PR TITLE
[baggage-processor] Add BaggageLogRecordProcessor

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
**Description:**

Support promoting Baggage attributes as log attributes the same we we can promote baggage attributes as span attributes.

**Existing Issue(s):**

* https://github.com/open-telemetry/opentelemetry-java-contrib/issues/1577

**Testing:**

Enriched the `BaggageProcessorCustomizerTest` with tests for log entries.

**Documentation:**

README.md updated

**Outstanding items:**

We can enrich metrics with Baggage attributes the same way we enrich spans and logs because there is no equivalent for metrics to the [Span Processor API](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.39.0/specification/trace/sdk.md#span-processor) and the [Log Record PRocessor API](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.39.0/specification/logs/sdk.md#logrecordprocessor) in OTel SDK specifications.

### CODE REVIEW

File: baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java [hunk 1]
```diff
@@ -38,28 +43,35 @@
...
     InMemorySpanExporter spanExporter = InMemorySpanExporter.create();
+    InMemoryLogRecordExporter logExporter = InMemoryLogRecordExporter.create();
 
-    OpenTelemetrySdk sdk = getOpenTelemetrySdk(properties, spanExporter);
+    OpenTelemetrySdk sdk = getOpenTelemetrySdk(properties, spanExporter, logExporter);
     try (Scope ignore = Baggage.current().toBuilder().put("key", "value").build().makeCurrent()) {
       sdk.getTracer("test").spanBuilder("test").startSpan().end();
+      sdk.getLogsBridge().get("test").logRecordBuilder().setBody ("test").emit();
     }
+    // TODO verify log record attributes
```
Comment (cyrille-leclerc) [inline baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:74]: @zeitlinger can you please help me write a test to check the `key` baggage attribute has been promoted as a log attribute?

Comment (zeitlinger) [reply baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:73]: sure: https://github.com/cyrille-leclerc/opentelemetry-java-contrib/pull/1

File: baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java [hunk 2]
```diff
@@ -38,57 +45,69 @@
...
     assertCustomizer(Collections.emptyMap(), span -> span.hasTotalAttributeCount(0));
-    assertCustomizer(
-        Collections.singletonMap(
-            "otel.java.experimental.span-attributes.copy-from-baggage.include", "key"),
-        span -> span.hasAttribute(AttributeKey.stringKey("key"), "value"));
+    Map<String, String> properties = new HashMap<>();
+    properties.put("otel.java.experimental.span-attributes.copy-from-baggage.include", "key");
+    properties.put("otel.java.experimental.log-attributes.copy-from-baggage.include", "key");
+    // TODO try use
+    //  AttributeAssertion attributeAssertion =
```
Comment (cyrille-leclerc) [inline baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:59]: @zeitlinger do you know if we could use an ÀttributeAssertion`on both span and log record data? I couldn't find a solution.

File: baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java [hunk 3]
```diff
@@ -38,57 +45,69 @@
...
-            () ->
-                TracesAssert.assertThat(spanExporter.getFinishedSpanItems())
-                    .hasTracesSatisfyingExactly(
-                        trace -> trace.hasSpansSatisfyingExactly(spanDataAssertConsumer)));
+            () -> {
+              TracesAssert.assertThat(spanExporter.getFinishedSpanItems())
+                  .hasTracesSatisfyingExactly(
+                      trace -> trace.hasSpansSatisfyingExactly(spanDataAssertConsumer));
+              List<LogRecordData> finishedLogRecordItems = logExporter.getFinishedLogRecordItems();
+                  assertThat(finishedLogRecordItems).hasSize(1);
```
Comment (cyrille-leclerc) [inline baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:84]: @zeitlinger is it enough testing of the log record?

File: baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java (hunk 2 above)
Comment (zeitlinger) [reply baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:59]:
```java
 assertThat(finishedLogRecordItems)
                  .hasSize(1)
                  .anySatisfy(
                      logRecord -> OpenTelemetryAssertions.assertThat(logRecord.getAttributes())
                          .containsEntry(AttributeKey.stringKey("key"), "value"));
```

File: baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java (hunk 3 above)
Comment (zeitlinger) [reply baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java:84]: no, see last comment

File: baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizer.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (keys.isEmpty()) {
+      return;
+    }
+
+    sdkTracerProviderBuilder.addSpanProcessor(createBaggageSpanProcessor(keys));
+  }
+
+  static BaggageSpanProcessor createBaggageSpanProcessor(List<String> keys) {
+    if (keys.size() == 1 && keys.get(0).equals("*")) {
```
Comment (zeitlinger) [inline baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizer.java:44]: nit: extract `isWildcard` method to be reused for logger

**Review by zeitlinger — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Pull request merged**